pydrive2/auth.py

```diff
@@ -523,7 +524,7 @@ def Authorize(self):
     """
     if self.http is None:
       self.http = httplib2.Http(timeout=self.http_timeout)
-    if self.access_token_expired:
+    if self.access_token_expired and self.auth_method != 'service':
```

Comment:
same here, not sure I understand why do we need this second check

Reply:
"For the Service account we don't want to Refresh the access token, since refresh token is not available for Service accounts. We just want to Auth as usual and request new access token."

Reply:
do we need a p12 file around for that?

Reply:
at this stage the content of p12 file already should be loaded into `self.credentials`. If it was loaded from `gdrive-user-credentials.json` then p12 file is not expected to be used and the path is not verified.